src/monsters.cpp

```diff
@@ -1187,21 +1151,14 @@ bool Monsters::loadMonster(const std::string& file, const std::string& monster_n
 	if ((node = monsterNode.child("script"))) {
 		for (pugi::xml_node eventNode = node.first_child(); eventNode; eventNode = eventNode.next_sibling()) {
 			if ((attr = eventNode.attribute("name"))) {
-				mType->scriptList.push_back(attr.as_string());
+				mType->scriptList.emplace_back(attr.as_string());
 			} else {
 				std::cout << "[Warning - Monsters::loadMonster] Missing name for script event. " << file << std::endl;
 			}
 		}
 	}
 
-	static uint32_t id = 0;
-	if (new_mType) {
-		std::string lowername = monster_name;
-		toLowerCaseString(lowername);
-
-		monsterNames[lowername] = ++id;
-		monsters[id] = mType;
-	}
+	mType->shrinkToFit();
```

Comment:
Removed shrinkVectors, however, I still think it's better to keep the MonsterType::shrinkToFit() method - Monsters::loadMonster() is long enough already ;).